js/js.go

```diff
@@ -153,6 +153,103 @@ func MakeWrapper(i interface{}) *Object {
 	return o
 }
 
+// MakeFullWrapper creates a JavaScript object which has wrappers for the exported
+// methods of i, and, where i is a (pointer to a) struct value, wrapped getters
+// and setters
+// (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/defineProperty)
+// for the non-embedded exported fields of i. Values accessed via these methods
+// and getters are themsevles wrapped when accessed, but an important point to
+// note is that a new wrapped value is created on each access.
+func MakeFullWrapper(i interface{}) *Object {
+	v := InternalObject(i)
+	c := v.Get("constructor")
+
+	o := Global.Get("Object").New()
+
+	defineProperty := func(k string, fns ...func(*Object)) {
+		op := Global.Get("Object").New()
+		for _, f := range fns {
+			f(op)
+		}
+		Global.Get("Object").Call("defineProperty", o, k, op)
+	}
+
+	defineProperty("__internal_object__", func(op *Object) {
+		op.Set("value", v)
+	})
+
+	{
+		// caculate a sensible type string
+
+		// we don't want to import any packages in this package
+		// so we do some string operations by hand
```

Comment:
readability: Comments should be full sentences starting with a capital letter and ending with punctuation. Same below.